src/main/kotlin/org/jetbrains/intellij/platform/gradle/plugins/project/IntelliJPlatformModulePlugin.kt

```diff
@@ -38,18 +39,59 @@ abstract class IntelliJPlatformModulePlugin : Plugin<Project> {
             apply(IntelliJPlatformBasePlugin::class)
         }
 
+        // To understand what is going on below read & watch this:
+        // https://youtu.be/2gPJD0mAres?t=461
+        // https://www.youtube.com/watch?v=8z5KFCLZDd0
+        // https://docs.gradle.org/current/userguide/variant_attributes.html#sec:standard_attributes
+        // https://docs.gradle.org/current/userguide/cross_project_publications.html#sec:variant-aware-sharing
         with(project.configurations) configurations@{
-            fun Configuration.applyVariantAttributes() {
+            fun Configuration.applyVariantCommonAttributes(customAction: AttributeContainer.() -> Unit = {}) {
                 attributes {
                     attribute(Bundling.BUNDLING_ATTRIBUTE, project.objects.named(Bundling.EXTERNAL))
                     attribute(Category.CATEGORY_ATTRIBUTE, project.objects.named(Category.LIBRARY))
+
+                    // https://github.com/JetBrains/intellij-platform-gradle-plugin/issues/1772
+                    // https://docs.gradle.org/current/userguide/cross_project_publications.html#targeting-different-platforms
+                    // > By default, the org.gradle.jvm.version is set to the value of the release property
+                    // > (or as fallback to the targetCompatibility value) of the main compilation task of the source set.
                     attributeProvider(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, project.provider {
                         project.the<JavaPluginExtension>().targetCompatibility.majorVersion.toInt()
                     })
-                    attribute(Usage.USAGE_ATTRIBUTE, project.objects.named(Usage.JAVA_RUNTIME))
+
                     attributes.attribute(Attributes.jvmEnvironment, "standard-jvm")
                     attributes.attribute(Attributes.kotlinJPlatformType, "jvm")
+
+                    customAction()
+                }
+            }
+
+            create(
+                name = Configurations.INTELLIJ_PLATFORM_COMPOSED_JAR_API,
+                description = "IntelliJ Platform final composed Jar archive Api",
+            ) {
+                // true & true is deprecated.
+                // https://docs.gradle.org/current/userguide/declaring_dependencies_adv.html#sec:resolvable-consumable-configs
+                // > For backwards compatibility, both flags have a default value of true, but as a plugin author,
+                // you should always determine the right values for those flags, or you might accidentally introduce
+                // resolution errors.
+                isCanBeConsumed = true
+                isCanBeResolved = false
```

Comment:
I made it consumable instead of true + true like INTELLIJ_PLATFORM_COMPOSED_JAR below.